### PR TITLE
Funding source create refactor

### PR DESCRIPTION
**Pull request by markgdawson**
Refactoring the code to implement #62

### CODE REVIEW

File: funding/forms.py [hunk 1]
```diff
@@ -47,7 +43,6 @@ def clean_pi_email(self):
         self.institution = Institution.objects.get(base_domain=domain)
         return email
 
-
```
Comment (edbennett) [inline funding/forms.py]: PEP 8 suggests "Surround top-level function and class definitions with two blank lines", so I wouldn't remove this

File: funding/tests/test_views.py [hunk 2]
```diff
@@ -75,20 +75,42 @@ def test_fundingsource_view_as_an_authorised_user(self):
...
+            self.assertEqual(response.context_data.get('form').initial['identifier'], test_identifier)
 
     def test_view_as_an_unauthorised_user(self):
         """
         Ensure unauthorised users can not access the project create view.
         """
+        # test endpoint with no identifier
         self.access_view_as_unauthorised_user(reverse('create-funding-source'))
 
+        # test endpoint with no identifier
```
Comment (edbennett) [inline funding/tests/test_views.py:110]: comment says "no identifier" but `args` say "some-identifier"?

File: funding/forms.py (hunk 1 above)
Comment (markgdawson) [reply funding/forms.py]: I just fixed this somewhere else, so I'm now "net zero" on PEP 8 changes.

Thanks Ed.

File: funding/urls.py [hunk 3]
```diff
@@ -9,10 +9,15 @@
...
-        'create-funding-source/',
+        'create-funding-source/<identifier>',
         views.FundingSourceCreateView.as_view(),
         name='create-funding-source',
     ),
+    path(
+        'create-funding-source/<str:identifier>',
+        views.FundingSourceCreateView.as_view(),
+        name='create-funding-source-with-identifier',
+    ),
```
Comment (edbennett) [inline funding/urls.py:20]: One of these is called `create-funding-source-with-identifier` and one is called `create-funding-source`, but they both take identifiers?

**Pull request merged**